src/components/ResultContainer.tsx

```diff
@@ -0,0 +1,220 @@
+      }
+
+      return tsvRows.join('\n');
+    }
+
+    return '';
+  }
+
+  function downloadResults(buttonIdentifier: string) {
+    const element = document.createElement('a');
+    const encodedTSV = encodeURIComponent(generateTSVString(buttonIdentifier));
+    element.setAttribute('href', `data:text/tab-separated-values;charset=utf-8,${encodedTSV}`);
+    element.setAttribute('download', `${buttonIdentifier}-results.tsv`);
+
+    element.style.display = 'none';
+    document.body.appendChild(element);
+
+    element.click();
+    document.body.removeChild(element);
+  }
+
+  function renderResults() {
+    if (result === null) {
+      return (
+        <Typography variant="h5" className="text-gray-500">
+          Click &apos;Submit Query&apos; for results
+        </Typography>
+      );
+    }
+    if (result.length === 0) {
+      return (
+        <Typography variant="h5" className="text-gray-500">
+          No results
+        </Typography>
+      );
+    }
+    return (
+      <>
+        <div className="col-span-1 row-start-2">
+          <FormControlLabel
+            label="Select all datasets"
+            control={
+              <Checkbox
+                onChange={(event) => handleSelectAll(event.target.checked)}
+                checked={selectAll()}
+              />
+            }
+          />
+        </div>
+        <div className="col-span-4 max-h-96 space-y-2 overflow-auto">
+          {result.map((item) => (
+            <ResultCard
+              key={item.dataset_uuid}
+              nodeName={item.node_name}
+              datasetUUID={item.dataset_uuid}
+              datasetName={item.dataset_name}
+              datasetTotalSubjects={item.dataset_total_subjects}
+              numMatchingSubjects={item.num_matching_subjects}
+              imageModals={item.image_modals}
+              checked={download.includes(item.dataset_uuid)}
+              onCheckboxChange={(checked, id) => updateDownload(checked, id)}
+            />
+          ))}
+        </div>
+        <div className="col-span-1">
+          <Button variant="contained" onClick={() => setOpenModal(true)}>
+            How to get data
+          </Button>
+          <NBDialog open={openModal} onClose={() => setOpenModal(false)} />
+        </div>
+        <div className="col-span-3 space-x-2 justify-self-end">
+          <DownloadResultButton
+            identifier="participant-level"
+            disabled={download.length === 0}
+            handleClick={() => downloadResults('participant-level')}
+          />
+          <DownloadResultButton
+            identifier="dataset-level"
+            disabled={download.length === 0}
+            handleClick={() => downloadResults('dataset-level')}
+          />
+        </div>
+      </>
+    );
+  }
+
+  return (
+    <div className="grid grid-cols-4 grid-rows-2">
+      <div className="col-span-4">
+        <Typography variant="h5">Results</Typography>
+      </div>
+      {renderResults()}
```

Comment:
Not sure I fully understand why `renderResults` is a function. Could you explain why the logic from `renderResults` can't just live here directly?

Reply:
Eslint didn't like nesting if statements inside JSX which was the case when the code lived directly there and we had to use conditional rendering.

Reply:
OK. But we should understand what the intention of the eslint rule is that we were breaking before (and with the I assume more intuitive syntax). Maybe there is a good way to make eslint happy here without abstracting it all away. Could you put the rule here that was upset?